### ISSUE TITLE
[v1.7.1] Release Tracker

### ISSUE THREAD

**Issue by seemethere** [pytorch/pytorch#47622]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/1.7) for the 1.7.1 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 11/18): work on finalizing the release branch
* Phase 2 (after 12/10): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 11/18):**

Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent minor release (e.g. 1.7.0 for 1.7.x release; see [topic: regression issue list](http://github.com/pytorch/pytorch/issues?q=is:issue%20is:open%20label:%22topic:%20regression%22/))
2. Critical fixes for: [silent correctness](http://github.com/pytorch/pytorch/issues?q=is:issue%20is:open%20label:%22topic:%20correctness%20(silent)"/), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](http://is:issue%20is:open%20label:%22topic:%20deadlock"/), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+).
3. Fixes to new features introduced in the most recent minor release (e.g. 1.7.0 for 1.7.x release)
4. Test/CI fixes
5. Documentation improvements
6. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently [@malfet](https://github.com/malfet), [@seemethere](https://github.com/seemethere), [@gchanan](https://github.com/gchanan)).   If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 12/10):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are:

**Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

This will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/1.7).
3. Make a request below with the following format:

```
`Link to landed master PR (if applicable):
Link to release branch PR:
Criteria Category:`
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **NOTE: do not land the release branch PR yourself. Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

**Comment by hameerabbasi** [comment]
Link to landed master PR: #47110
Link to release branch PR: #47735
Criteria Category: silent correctness

**Comment by hameerabbasi** [comment]
Link to landed master PR: #45983
Link to release branch PR: #47736
Criteria Category: silent correctness

**Comment by hameerabbasi** [comment]
Link to non-landed master PR: #47842
Link to release branch PR: #47843
Criteria Category: documentation bug

There is issue #47697 labelled high-prio and tagged with the 1.7.1 milestone.

**Comment by hameerabbasi** [comment]
@gchanan Would you know anything about unrelated CI failures on all of my fixes?

**Comment by seemethere** [comment]
Adding Python 3.9 support to `v1.7.1`
----

Link to landed master PR:https://github.com/pytorch/pytorch/pull/47235, https://github.com/pytorch/pytorch/pull/47689
Link to release branch PR: https://github.com/pytorch/pytorch/pull/48133
Criteria Category: Special Dispensation

**Comment by bertmaher** [comment]
I've a collection of hotfixes for the tensorexpr fuser: 

Release branch PR: #48137
Master PRs are linked in the branch PR
Criteria: new feature fix; silent correctness or crash

**Comment by bertmaher** [comment]
@seemethere is there anything more I have to do to get the TE bugfix PR in?

**Comment by seemethere** [comment]
Various setup.py fixes (dataclasses, minimum python version)
------
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46921 https://github.com/pytorch/pytorch/pull/45844 https://github.com/pytorch/pytorch/pull/47314
Link to release branch PR: https://github.com/pytorch/pytorch/pull/48220
Criteria Category: Special Dispensation

**Comment by gchanan** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/47772
Link to release branch PR: https://github.com/pytorch/pytorch/pull/48215
Criteria Category: Fixes to regressions against the most recent minor release

**Comment by gchanan** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/47424
Link to release branch PR: https://github.com/pytorch/pytorch/pull/48216
Criteria Category: Critical fixes for: silent correctness, crashes, deadlocks, (large) memory leaks.

**Comment by gchanan** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/47065
Link to release branch PR: https://github.com/pytorch/pytorch/pull/48219
Criteria Category: Fixes to regressions against the most recent minor release

**Comment by seemethere** [comment]
> @gchanan Would you know anything about unrelated CI failures on all of my fixes?

All of your PRs were actually done against the wrong base branch, they should've been done against the `release/1.7` base branch.

I don't know who created the `1.7` branch but it's not the release branch

**Comment by seemethere** [comment]
@hameerabbasi I went through and fixed all of your PRs, going to let them run through CI

**Comment by hameerabbasi** [comment]
> @hameerabbasi I went through and fixed all of your PRs, going to let them run through CI

Thanks, and sorry for the confusion, I was in the process of fixing them myself.

**Comment by malfet** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/48151
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48309
Criteria Category: Fixes to regressions against the most recent minor release

**Comment by malfet** [comment]
Link to landed master PRs:  https://github.com/pytorch/pytorch/pull/47398 
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48311 https://github.com/pytorch/pytorch/pull/47752 https://github.com/pytorch/pytorch/pull/48076
Criteria Category: Fixes to regressions against the previous major release

**Comment by seemethere** [comment]
Pybind11 update for python 3.9 compat
----

Link to landed master PR: https://github.com/pytorch/pytorch/pull/48117
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48312
Criteria Category: Special Dispensation

**Comment by malfet** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46317
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48317
Criteria Category: Fixes to regressions against the previous major release

**Comment by malfet** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/47006
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48319
Criteria Category: Fixes to regressions against the previous minor release

**Comment by malfet** [comment]
Link to landed master PRs: https://github.com/pytorch/pytorch/pull/47262 https://github.com/pytorch/pytorch/pull/47337 and https://github.com/pytorch/pytorch/pull/47390
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48337
Criteria Category: Fixes latent bug against previous release

**Comment by malfet** [comment]
Link to landed master PR: 
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48744

**Comment by seemethere** [comment]
Link to landed master pr: https://github.com/pytorch/pytorch/issues/48618
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48768
Criteria Category: Special dispensation (fixes cpp extension building for windows for Python 3.9)

**Comment by mrshenli** [comment]
Placeholder for NCCL all2all, cc @mingzhe09088 @msbaines

**Comment by mingzhe09088** [comment]
Link to laned master PR: #45900
Link to cherry pick PR: #48857
Criteria Category: Enable NCCL A2A on OSS (fixes duplicate NCCL linking issue)

**Comment by Kyeongpil** [comment]
Link to release branch PR: #48775
Criteria Category: Documentation improvements

**Comment by seemethere** [comment]
Link to landed master pr: https://github.com/pytorch/pytorch/pull/47150
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48937
Criteria Category: Special dispensation (fixes cpp extension building for windows for Python 3.9)

**Comment by seemethere** [comment]
Link to landed master pr: https://github.com/pytorch/pytorch/pull/48696
Link to cherry pick PR: https://github.com/pytorch/pytorch/pull/48936
Criteria Category: Fixes memory leak

**Comment by walterddr** [comment]
Link to landed master PR: ##48590
Link to cherry pick PR: #48943
Criteria Category: fix flake8

**Comment by seemethere** [comment]
1.7.1 has been released, going to go ahead and close / unpin this issue
